From 98234acd5b48a58c2d112b5edb3dddb85e04a643 Mon Sep 17 00:00:00 2001
From: Daniel van Vugt <daniel.van.vugt@canonical.com>
Date: Fri, 2 Oct 2020 08:33:31 +0000
Subject: [PATCH] windowManager: Avoid calling meta_window_actor_thaw when not
 frozen

Because that tends to crash in:
```
g_error ("Error in freeze/thaw accounting");
```

Closes: https://gitlab.gnome.org/GNOME/mutter/-/issues/1431

https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1453


(cherry picked from commit 6ba3ca5f9548ba0a87a9e3ffa129716d0a062e1e)
---
 js/ui/windowManager.js | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/js/ui/windowManager.js b/js/ui/windowManager.js
index ab86a2e6a3..91b764ea8d 100644
--- a/js/ui/windowManager.js
+++ b/js/ui/windowManager.js
@@ -1378,6 +1378,10 @@ var WindowManager = class {
             onStopped: () => this._sizeChangeWindowDone(shellwm, actor),
         });
 
+        // ease didn't animate and cleared the info, we are done
+        if (!actor.__animationInfo)
+            return;
+
         // Now unfreeze actor updates, to get it to the new size.
         // It's important that we don't wait until the animation is completed to
         // do this, otherwise our scale will be applied to the old texture size.
-- 
GitLab

